fsspec/parquet.py

```diff
@@ -119,8 +163,24 @@ def _get_parquet_byte_ranges(
     `KnownPartsOfAFile` caching strategy of a single path.
     """
 
-    # Set the engine
-    engine = _set_engine(engine)
+    # Pass to specialized function if metadata is defined
+    if metadata is not None:
+
+        # Use the provided parquet metadata object
+        # to avoid transfering/parsing footer metadata
+        return _get_parquet_byte_ranges_from_metadata(
+            metadata,
```

Comment:
Derive value of engine from the type of the metadata? Will get weird error on mismatch. Not urgent.

Reply:
Yes - We absolutely should do this, but I'd prefer to do it in a stand-alone PR if possible.